In [ ]:
#Import necessory libraries
import selenium
import pandas as pd
import time
import  requests
from selenium import webdriver
from selenium.common.exceptions import NoSuchElementException
import re

import warnings
warnings.filterwarnings('ignore')
%matplotlib inline

In [ ]:
#lets import input data
df = pd.read_csv(r"Input.xlsx - Sheet1.csv")
df.head(10)

URL_ID                                                URL
0       1  https://insights.blackcoffer.com/how-is-login-...
1       2  https://insights.blackcoffer.com/how-does-ai-h...
2       3  https://insights.blackcoffer.com/ai-and-its-im...
3       4  https://insights.blackcoffer.com/how-do-deep-l...
4       5  https://insights.blackcoffer.com/how-artificia...
5       6  https://insights.blackcoffer.com/how-are-genet...
6       7  https://insights.blackcoffer.com/how-is-ai-use...
7       8  https://insights.blackcoffer.com/benefits-of-b...
8       9  https://insights.blackcoffer.com/how-big-data-...
9      10  https://insights.blackcoffer.com/how-will-ai-m...

In [ ]:
#lets check the shape of our data 
print("Shape of the input data :", df.shape)

Shape of the input data : (170, 2)


Great! we are having arround 170 urls to fetch the text from them.

In [ ]:
#Lets define our web-driver to brows the data
driver = webdriver.Chrome(r"C:\Users\ABHINANDAN\Desktop\chromedriver.exe")

In [ ]:
#lets define a function to fetch Article title and text
def text_extracter(url):
    #getting article page
    driver.get(url)
    time.sleep(2)
    title = driver.find_element_by_xpath("//h1[@class='entry-title']")  #finding article title
    article_title.append(title.text)
    
    #getting the article text
    txt = driver.find_element_by_xpath("//div[@class='td-post-content']")
    article_text.append(txt.text.replace('\n',' '))

In [ ]:
#lets extract the text from every link in our data set
article_title=[]
article_text=[]
for url in df.URL:
    text_extracter(url)
    
#lets print the length to confirm
print(len(article_title),len(article_text))

170 170


In [ ]:
df["Article_Title"]=article_title
df["Article_Text"] = article_text

In [ ]:
df

URL_ID                                                URL  \
0         1  https://insights.blackcoffer.com/how-is-login-...   
1         2  https://insights.blackcoffer.com/how-does-ai-h...   
2         3  https://insights.blackcoffer.com/ai-and-its-im...   
3         4  https://insights.blackcoffer.com/how-do-deep-l...   
4         5  https://insights.blackcoffer.com/how-artificia...   
..      ...                                                ...   
165     167  https://insights.blackcoffer.com/role-big-data...   
166     168  https://insights.blackcoffer.com/sales-forecas...   
167     169  https://insights.blackcoffer.com/detect-data-e...   
168     170  https://insights.blackcoffer.com/data-exfiltra...   
169     171  https://insights.blackcoffer.com/impacts-of-co...   

                                         Article_Title  \
0    How is Login Logout Time Tracking for Employee...   
1    How does AI help to monitor Retail Shelf watches?   
2            AI and its impact on the Fashion Industry   
3    How do deep learning models predict old and ne...   
4    How artificial intelligence can boost your pro...   
..                                                 ...   
165                       Role of Big Data in Academia   
166  Statistical Methods for Sales Forecasting in R...   
167  Behavior Based Chi-Square model to Detect Data...   
168                         What is Data Exfiltration?   
169               Impacts of COVID 19 on Food products   

                                          Article_Text  
0    When people hear AI they often think about sen...  
1    With increasing computing power and more data,...  
2    If you were a fan of the 90’s film Clueless ba...  
3    Understanding exactly how data is ingested, an...  
4    From the stone age to the modern world, from h...  
..                                                 ...  
165  CAN ACADEMIA, RESEARCHERS, DECISION MAKERS AND...  
166  Introduction Inventory planning is a fundament...  
167  1. The Problem Insider threat detection specif...  
168  If we talk in terms of our general life, Exfil...  
169  Some vendors (fruit and vegetable sellers) beg...  

[170 rows x 4 columns]

In [ ]:
#saving the data into csv file
df.to_csv(r"C:\Users\ABHINANDAN\Desktop\Blackcoffer_article.csv")